commons/src/main/java/info/metadude/android/eventfahrplan/commons/DateFormatter.kt

```diff
@@ -0,0 +1,51 @@
+package info.metadude.android.eventfahrplan.commons
+
+import org.threeten.bp.OffsetDateTime
+import org.threeten.bp.format.DateTimeFormatter
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * Format timestamps according to system locale and system time zone.
+ * */
+class DateFormatter {
+    private val timeShort = SimpleDateFormat.getTimeInstance(SimpleDateFormat.SHORT)
+    private val timeShortNumberOnly = DateTimeFormatter.ofPattern("HH:mm")
+    private val dateShort = SimpleDateFormat.getDateInstance(SimpleDateFormat.SHORT)
+    private val dateTimeShort = SimpleDateFormat.getDateTimeInstance(SimpleDateFormat.SHORT, SimpleDateFormat.SHORT)
+    private val dateTimeFull = SimpleDateFormat.getDateTimeInstance(SimpleDateFormat.FULL, SimpleDateFormat.SHORT)
+
+    /*
+    * Returns 01:00, 14:00 etc for all locales. Always without AM or PM postfix and always in 24 hour format.
+    * */
+    fun getFormattedTimeNumbersOnly(clock: SystemClock): String {
+        return timeShortNumberOnly.format(clock.toZonedDateTime(OffsetDateTime.now().offset))
+    }
+
+    /*
+    * Returns 01:00 AM, 02:00 PM, 14:00 etc.
+    * Postfix AM or PM (and 24 hour format) is used depending on current system locale.
+    * */
+    fun getFormattedTime(time: Long): String {
+        return timeShort.format(Date(time))
+    }
+
+    fun getFormattedDate(time: Long): String {
+        return dateShort.format(Date(time))
+    }
+
+    fun getFormattedDateTime(time: Long): String {
+        return dateTimeFull.format(Date(time))
+    }
+
+    fun getFormattedDateTimeShort(time: Long): String {
+        return dateTimeShort.format(Date(time))
+    }
+
+    companion object {
+
+        fun get(): DateFormatter {
+            return DateFormatter();
+        }
```

Comment:
I added now a private primary constructor. I like the static factory method pattern, especially since in Kotlin "new" is not required anymore, now in Kotlin and Java DateFormatter instance retrieval looks the same.

Reply:
A common naming pattern to retrieve a new instance via a static factory method is `newInstance()`. You can find it being used across the project. For consistency reason I would recommend to use this method name.

Reply:
Done with 8bec912.